src/ext/minicylc.py

```diff
@@ -0,0 +1,254 @@
+
+from sphinx.ext.graphviz import (
+    graphviz, GraphvizSimple, html_visit_graphviz, latex_visit_graphviz,
+    texinfo_visit_graphviz, text_visit_graphviz, man_visit_graphviz)
+
+
+class MiniCylc(graphviz):
+    """Node to represent an animated Cylc graph.
+
+    Works by extending the built-in ``sphinx.ext.graphviz`` code.
+
+    """
+
+    @staticmethod
+    def promote_graphviz(graphviz_node):
+        """Promote a sphinx.ext.graphviz node to a MiniCylc node.
+
+        Returns a new MiniCylc node with attributes set to the values of the
+        provided graphviz node.
+
+        """
+        # Duplicate provided node
+        new_node = MiniCylc()
+        for item in dir(graphviz_node):
+            if item.startswith('__'):
+                # Skip builtins.
+                continue
+            attr = getattr(graphviz_node, item)
+            if isinstance(attr, types.MethodType):
+                # Skip methods.
+                continue
+            setattr(new_node, item, attr)
+
+        # Use the digraph graphviz layout.
+        new_node['code'] = new_node['code'].replace('minicylc', 'digraph')
+        return new_node
+
+    @staticmethod
+    def visit_html(builder, node):
+        """Wrap the standard graphviz output to provide alignment for svg.
+
+        At present the ``align`` option for ``sphinx.ext.graphviz`` directives
+        does nothing for html output when ``graphviz_output_format`` is set to
+        ``svg``. This method wraps the html output to provide alignment
+        capability.
+
+        """
+        # Alignment.
+        try:
+            align = node['align']
+        except KeyError:
+            align = 'center'
+        style = 'text-align: %s;' % align
+
+        # Theme.
+        theme = node.get('theme', 'default')
+
+        # Graphing.
+        data = '//'.join(node['graph_lines'])
+
+        # Construct <div> element.
+        builder.body.append(
+            ('<div class="minicylc" style="{0}" data-dependencies="{1}" '
+             'data-theme="{2}" >').format(style, data, theme))
+
+        # Call graphviz html builder.
+        try:
+            # This method raises nodes.SkipNode in the success and fail case!
+            html_visit_graphviz(builder, node)
+        except nodes.SkipNode:
+            # Close <div> element (we are not using an exit_html function.
+            builder.body.append('</div>')
+            raise
+
+
+class MiniCylcDirective(GraphvizSimple):
+    """Implement the ``mini-cylc`` directive for animating Cylc graphs.
+
+    Works by extending the built-in ``sphinx.ext.graphviz`` code.
+
+    """
+    option_spec = dict(GraphvizSimple.option_spec)
+    option_spec['snippet'] = directives.flag
+    option_spec['theme'] = directives.unchanged
+    option_spec['size'] = directives.unchanged
+    required_arguments = 0  # Arg will be provided in run().
+
+    CONDITIONAL_CHARS = ['&', '|', '(', ')']
+    CONDITIONAL_REGEX = re.compile(r'([\(\)\|\&])')
```

Comment:
Same as `CONDITIONAL_REGEX = re.compile(r'([()|&])')` ? (no need to escape these characters when using `[]`?)